cloudinit/config/schema.py

```diff
@@ -474,15 +533,7 @@ def handle_schema_args(name, args):
                 cfg_name = args.config_file
             print("Valid cloud-config:", cfg_name)
     elif args.docs:
-        schema_ids = [subschema['id'] for subschema in full_schema['allOf']]
-        schema_ids += ['all']
-        invalid_docs = set(args.docs).difference(set(schema_ids))
-        if invalid_docs:
-            error('Invalid --docs value {0}. Must be one of: {1}'.format(
-                  list(invalid_docs), ', '.join(schema_ids)))
-        for subschema in full_schema['allOf']:
-            if 'all' in args.docs or subschema['id'] in args.docs:
-                print(get_schema_doc(subschema))
+        print(load_doc(args.docs))
```

Comment:
So, with this change you have dropped the ability to specify individual schema modules on the commandline via:
```PYTHONPATH=. python3 -m cloudinit.cmd devel schema --docs cc_write_files``` while this may be an under-utilized or under-documented feature of the command line, it does represent a change that we should probably discuss the merit of. 

Minimally, if we are dropping that ability to specify individual modules for documentation on the command line, we need to make sure our [argument parser is updated to avoid prompting for an `--all` value which is now disregarded](https://github.com/canonical/cloud-init/blob/e63069297ba5d75e47c1ed1149dd6cd3f1ac9a92/cloudinit/config/schema.py#L507).

Reply:
That command doesn't work for me, but in a container [I see this](https://gist.github.com/holmanb/b69e142a2b6be5729a1f7b7bf4e0434c), which is what I would expect for specifying individual modules to print out. Is there something about running it as a module that causes different behavior perhaps? The logic for selecting which modules to print was moved into `load_doc()` and I think it should still work. The functional change was to concatenate module docstrings rather than call `get_schema_doc()`.

Reply:
I just noticed that load_doc() isn't doing the same error checking which causes a behavioral change,. I think that this expected behavior probably warrants a unittest. I'll fix it and add one.

Reply:
@blackboxsw - I just pushed some [tests to verify expected behavior](https://github.com/canonical/cloud-init/pull/1101/commits/662564e19994df3e185d3506a7017bbcd2e30fab).